inc/class-wp-cli-command.php

```diff
@@ -99,6 +99,39 @@ public function save( $args, $args_assoc ) {
 			if ( ! empty( $args_assoc['verbose'] ) ) {
 				WP_CLI::line( 'Saving ' . $url );
 			}
+
+			$options = [
+				'user_id' => get_current_user_id(),
+				'context' => '',
+				'action'  => '',
+			];
+
+			$post_id = url_to_postid( $url );
+			if ( ! empty( $post_id ) ) {
+				$post = get_post( $post_id );
+				if ( $post->post_type === 'page' ) {
+					$options['context'] = 'page';
+					$options['action']  = 'wp_cli_netstorage_publish';
+				}
+			} else {
+				$parsed_url = wp_parse_url( $url );
+				if ( ! empty( $parsed_url ) ) {
+					$args = array(
+						'meta_key'   => 'netstorage_path',
+						'meta_value' => substr( $parsed_url['path'], 1 ),
+						'post_type'  => 'netstorage-file',
+					);
+
+					$query = new \WP_Query( $args );
+					if ( ! empty( $query->post ) ) { // Get the first post assuming no posts will have the same path.
+						$options['context'] = 'netstorage-file';
+						$options['action']  = 'wp_cli_netstorage_publish';
+					}
+				}
+			}
+
+			do_action( 'ns_wp_cli_export_page', $post_id, $options );
```

Comment:
Will this hook be able to work even if the `context` and `action` keys are empty?

If yes, then the code can stay as it its, if no, then the code needs to be changed to only do the action if the required data is present.

In addition, there is a check above for `if ( ! empty( $post_id ) )`, which leads me to believe that there could be no post id in the variable.

Reply:
> Will this hook be able to work even if the context and action keys are empty?

Yep

> In addition, there is a check above for if ( ! empty( $post_id ) ), which leads me to believe that there could be no post id in the variable.

Its an additional check to ensure post ID is set and not `0`